pkg/pipeline/BulkUpdateService.go

```diff
@@ -0,0 +1,248 @@
+			return nil, err
+		}
+		for _, app := range appsNotGlobal {
+			if _, AppAlreadyAddedToResponse := AppTrackMap[app.Id]; AppAlreadyAddedToResponse {
+				continue
+			}
+			AppTrackMap[app.Id] = app.AppName
+			impactedObject := &ImpactedObjectsResponse{
+				AppId:   app.Id,
+				AppName: app.AppName,
+				EnvId:   envId,
+			}
+			impactedObjectsResponse = append(impactedObjectsResponse, impactedObject)
+		}
+	}
+	return impactedObjectsResponse, nil
+}
+func (impl BulkUpdateServiceImpl) ApplyJsonPatch(patch jsonpatch.Patch, target string) (string, error) {
+	modified, err := patch.Apply([]byte(target))
+	if err != nil {
+		impl.logger.Error("error in applying JSON patch")
+		return "Patch Failed", err
+	}
+	return string(modified), err
+}
+func (impl BulkUpdateServiceImpl) BulkUpdateDeploymentTemplate(bulkUpdatePayload BulkUpdatePayload) (string, error) {
+	if len(bulkUpdatePayload.Includes.Names) == 0 || len(bulkUpdatePayload.Excludes.Names) == 0 {
+		return "Please don't leave includes.names/excludes.names array empty", nil
+	}
+	patchJson := []byte(bulkUpdatePayload.DeploymentTemplate.Spec.PatchJson)
+	patch, err := jsonpatch.DecodePatch(patchJson)
+	SuccessMessage := "Bulk Update is successful"
+	if err != nil {
+		impl.logger.Errorw("error in decoding JSON patch", "err", err)
+		return "The patch string you entered seems wrong, please check and try again", err
+	}
+	UpdatedPatchMap := make(map[int]string)
+	var charts []*chartConfig.Chart
+	if bulkUpdatePayload.Global {
+		charts, err = impl.bulkUpdateRepository.FindBulkChartsByAppNameSubstring(bulkUpdatePayload.Includes.Names, bulkUpdatePayload.Excludes.Names)
+		if err != nil {
+			impl.logger.Error("error in fetching charts by app name substring")
+			return "Internal server error", err
+		}
+		for _, chart := range charts {
+			modified, err := impl.ApplyJsonPatch(patch, chart.Values)
+			if err != nil {
+				impl.logger.Error("error in applying JSON patch")
+				return "Error in applying patch, please check your script again", err
+			}
+			UpdatedPatchMap[chart.Id] = modified
+		}
+		err = impl.bulkUpdateRepository.BulkUpdateChartsValuesYamlAndGlobalOverrideById(UpdatedPatchMap)
+		if err != nil {
+			impl.logger.Error("error in bulk updating charts")
+			return "Internal server error - Bulk Update Failed", err
+		}
+	}
+	var chartsEnv []*chartConfig.EnvConfigOverride
+	for _, envId := range bulkUpdatePayload.EnvIds {
+		chartsEnv, err = impl.bulkUpdateRepository.FindBulkChartsEnvByAppNameSubstring(bulkUpdatePayload.Includes.Names, bulkUpdatePayload.Excludes.Names, envId)
+		if err != nil {
+			impl.logger.Errorw("error in fetching charts(for env) by app name substring", "err", err)
+			return "Internal server error", err
+		}
+		for _, chartEnv := range chartsEnv {
+			modified, err := impl.ApplyJsonPatch(patch, chartEnv.EnvOverrideValues)
+			if err != nil {
+				impl.logger.Errorw("Error in applying patch, please check your script again", "err", err)
```

Comment:
this will result in return in case of first error but shouldnt we execute all and then return all errors? doesnt make sense to stop abrubptly and not informing user what happened.
Probably we can tell user, which failed and why and which got applied